pkg/streamingpromql/operators/range_vector_selector.go

```diff
@@ -111,17 +114,23 @@ func (m *RangeVectorSelector) fillBuffer(floats *types.FPointRingBuffer, histogr
 				return nil
 			}
 		case chunkenc.ValHistogram, chunkenc.ValFloatHistogram:
-			t, h := m.chunkIterator.AtFloatHistogram(nil)
-			if value.IsStaleNaN(h.Sum) || t < rangeStart {
+			t := m.chunkIterator.AtT()
+			if t < rangeStart {
 				continue
```

Comment:
If `t >= rangeStart`, won't we miss the point entirely on the next call to `fillBuffer()`, given it calls `m.chunkIterator.Next()`?

I wonder if it's better to always append the point, like we do for floats, for consistency (and to avoid issues like this).

Reply:
We do the same for floats (see line 100), however you might be right that we miss the point. For range vector queries we reset the buffer before each `NextStepSamples`. But we don't do that on rates. Might need some more thought.

Reply:
Oh sorry, I misread this - I thought this was `t > rangeEnd` for some reason. Ignore me 🙂